src/ElasticAppSearchClient.php

```diff
@@ -41,22 +43,32 @@ class ElasticAppSearchClient implements ElasticAppsearchClientInterface {
    * Constructs a new AppSearchClient object.
    */
   public function __construct(ClientInterface $http_client, ConfigFactoryInterface $config) {
-    $this->httpClient = $http_client;    
-    $this->config = $config->get('elastic_appsearch.appsearch');
-    $this->server = $this->config->get('api_endpoint');
-    $this->apikey = $this->config->get('api_private_key');
-    $this->engineName = $this->config->get('engine_name');
-    
-    if(!empty($this->server)){
-      $this->connect($this->server, $this->apikey, $this->engineName);
+
+  }
+
+  public function getInstance($server){
+
+    if(!empty($this->client)){
+      return $this->client;
     }
 
+    $clientBuilder = ClientBuilder::create(
+      $server->getHost(),
+      $server->getSecret()
+    );
+
+    $this->client = $clientBuilder->build();
+
+    return $this->client;
   }
 
-  public static function connect($server, $apikey, $engine){
+  public static function connect($server, $apikey, $engine = FALSE){
     $clientBuilder = ClientBuilder::create($server, $apikey );
     $client = $clientBuilder->build();
-    $engine = $client->getEngine($engine);
+    if($engine){
+      $engine = $client->getEngine($engine);
+    }
+    return $client;
```

Comment:
You are always returning `$client`? shouldn't it return the `$engine` if `$engine` variable is true?